util/buildkitskipper/hasher/hasher.go

```diff
@@ -30,6 +31,11 @@ func (h *Hasher) GetHash() []byte {
 	return h.h.Sum(nil)
 }
 
+func (h *Hasher) HashStatement(name string, args []string) {
+	h.HashString(name)
+	h.HashString(strings.Join(args, " "))
```

Comment:
will `[]string{"a", "b"}` and `[]string{"a b"}` produce different hashes?

Reply:
> They seem to produce the same hash.

We need to avoid cases like this -- any changes that the user makes to their Earthfile that could possibly propagate down to their container, should result in a different hash.

> Shall I use a different separator?

This will lead to the same issue. instead you'll need to come up with a more robust way to hash them to different values. That's why we were using `json.Marshal` in the `HashCommand` function.

Reply:
I was going to suggest a separator that cannot be used in the args, if one exists, but JSON is fine too. Updated.